src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -327,6 +322,103 @@ internal object ReadODataOrCustomInstanceAnnotationValue(string annotationName,
+                collectionValidator,
+                validateNullValue,
+                isTopLevelPropertyValue,
+                insideResourceValue,
+                propertyName,
+                isDynamicProperty).ConfigureAwait(false);
+            this.AssertRecursionDepthIsZero();
+
+            return nonEntityValue;
+        }
+
+        /// <summary>
+        /// Asynchronously reads the value of the instance annotation.
+        /// </summary>
+        /// <param name="propertyAndAnnotationCollector">The duplicate property names checker instance.</param>
+        /// <param name="name">The name of the instance annotation.</param>
+        /// <returns>A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains the value of the instance annotation.</returns>
+        internal Task<object> ReadCustomInstanceAnnotationValueAsync(PropertyAndAnnotationCollector propertyAndAnnotationCollector, string name)
+        {
+            Debug.Assert(propertyAndAnnotationCollector != null, "propertyAndAnnotationCollector != null");
+            Debug.Assert(!string.IsNullOrEmpty(name), "!string.IsNullOrEmpty(name)");
+
+            object propertyAnnotation;
+            string odataType = null;
+            if (propertyAndAnnotationCollector.GetODataPropertyAnnotations(name)
+                .TryGetValue(ODataAnnotationNames.ODataType, out propertyAnnotation))
+            {
+                odataType = ReaderUtils.AddEdmPrefixOfTypeName(ReaderUtils.RemovePrefixOfTypeName((string)propertyAnnotation));
+            }
+
+            return ReadODataOrCustomInstanceAnnotationValueAsync(name, odataType);
+        }
+
+        /// <summary>
+        /// Asynchronously reads built-in "odata." or custom instance annotation's value.
+        /// </summary>
+        /// <param name="annotationName">The annotation name.</param>
+        /// <param name="odataType">the odata.type value if exists.</param>
+        /// <returns>A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains the annotation value.</returns>
+        internal Task<object> ReadODataOrCustomInstanceAnnotationValueAsync(string annotationName, string odataType)
+        {
+            // If this term is defined in the model, look up its type. If the term is not in the model, this will be null.
+            IEdmTypeReference expectedTypeFromTerm = MetadataUtils.LookupTypeOfTerm(annotationName, this.Model);
+            return this.ReadNonEntityValueImplementationAsync(
+                odataType,
+                expectedTypeFromTerm,
+                null, /*propertyAndAnnotationCollector*/
+                null, /*collectionValidator*/
+                false, /*validateNullValue*/
+                false, /*isTopLevelPropertyValue*/
+                false, /*insideResourceValue Always pass false here to try read @odata.type annotation in custom instance annotations*/

```

Comment:
```suggestion
                propertyAndAnnotationCollector: null,
                collectionValidator: null,
                validateNullValue: false,
                isTopLevelPropertyValue: false,
                insideResourceValue: false, // Always pass false here to try read @odata.type annotation in custom instance annotations
```
Nit: Prefer named parameters to /* parameterName */
Probably a solutionwide discussion required for this.